src/test_report.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Jeny Sadadia
+# Author: Jeny Sadadia <jeny.sadadia@gmail.com>
+
+import os
+import sys
+
+import kernelci.config
+import kernelci.data
+from kernelci.cli import Args, Command, parse_opts
+import jinja2
+
+
+class TestReport:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.data.get_db(self._db_config, api_token)
+
+    def _create_summaries(self, groups):
+        def squash(item, max_length):
+            if len(item) > max_length:
+                n = (max_length - 3) / 2
+                m = max_length - 3 - n
+                item = item[:m] + '...' + item[-n:]
+            return item
+
+        columns = ["test", "result"]
+        rows = [
+            (
+                g['test_case'],
+                g['test_result'],
+            ) for g in groups
+        ]
+        squashed = [tuple(squash(item, 28) for item in row)for row in rows]
+
+        widths = [len(col) for col in columns]
+        for row in squashed:
+            row_widths = [len(col) for col in row]
+            widths = [max(col) for col in zip(widths, row_widths)]
+        fmt = " | ".join("{{:{w}}}".format(w=w) for w in widths)
+        sep = '-+-'.join('-' * w for w in widths)
+        headers = '\n'.join([fmt.format(*tuple(columns)), sep])
+        summaries = [fmt.format(*row) for row in squashed]
+
+        return headers, summaries
+
+    def run(self):
+        sub_id = self._db.subscribe('node')
+        self._print("Listening for test completion events")
+        self._print("Press Ctrl-C to stop.")
+
+        try:
+            while True:
+                sys.stdout.flush()
+                event = self._db.get_event(sub_id)
+                if event.data['op'] != 'updated':
+                    continue
+
+                obj = self._db.get_node_from_event(event)
+                if obj['name'] != 'check-describe':
+                    continue
+
+                self._print(f"Node updated: {obj}")
+
+                test_group = [{'test_case': obj['name'],
+                               'test_result': str(obj['status'])}]
+                summary_headers, summaries = self._create_summaries(test_group)
+
+                for group, summary in zip(test_group, summaries):
+                    group['summary'] = summary
+
+                revision_obj = obj['revision']
+                data = {
```

Comment:
I believe we can just pass `{'nodes': nodes, 'root': root}` to the template. The `root` can be used for the common fields e.g. `root.revision.tree` etc. and the `nodes` would be iterated to populate the table with `node.name` and `node.status`.